okio/src/test/java/okio/ReadUtf8LineTest.java

```diff
@@ -94,6 +94,20 @@
     }
   }
 
+  @Test public void readUtf8LineStrictWithLimits() throws IOException {
+    data.writeUtf8("ab\nde\n");
+    assertEquals("ab", source.readUtf8LineStrict(4));
+    assertEquals("de", source.readUtf8LineStrict(4));
+
+    data.writeUtf8("abcdef\n");
+    try {
+      source.readUtf8LineStrict(4);
+      fail();
+    } catch (EOFException expected) {
+      assertEquals("\\n not found: size=4 content=6162636465660a…", expected.getMessage());
```

Comment:
The byte dump is an existing feature, so I'll let Jesse and Jake weigh in on whether they want to keep that.

The previous implementation still follows: the bytes are copied from the source instead of consumed. Regardless, let's definitely add tests and javadocs to protect that behavior.